### PR TITLE
Moved from window geometry to client geometry;

### PR DESCRIPTION
**Pull request by ZappaBoy**
Hi everybody,
Using `xborder` I noticed that there is an annoying gap on all the "Jetbrains" products. 

image

So I decided to investigate and I discovered the error:
The problem is in the following line: https://github.com/deter0/xborder/blob/0a8e5c951806b0f30ddec59ce26677f7075cf9b2/xborders#L386

It seems that using `window.get_geometry()` can cause problems for some windows, so I replaced with `window.get_client_window_geometry()` and everything works fine for me.

image

In addition, I do some little refactoring in order to improve the code quality and readability.

Hope this can help.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,8 @@
...
-pycairo
\ No newline at end of file
+certifi==2022.9.24
+charset-normalizer==2.1.1
+idna==3.4
+pycairo==1.21.0
+PyGObject==3.42.2
+PyGObject-stubs==1.0.0
+requests==2.28.1
+urllib3==1.26.12
```
Comment (deter0) [inline requirements.txt:8]: What are all these for?

**Comment by deter0** [comment]
Will this fix #31 and #33 possibly #40?

File: requirements.txt (hunk 1 above)
Comment (ZappaBoy) [reply requirements.txt:8]: There dependencies packages are needed for me to run it on IDE while developing and debugging.

**Comment by ZappaBoy** [comment]
> Will this fix #31 and #33 possibly #40?

Surely this PR will fix #33 (Tested all jetbrains products).
I'm not sure about #31 and #40 but I can confirm that I noticed some improvements in other windows (e.g. Yakuake) but still some little gaps remains.

**Comment by deter0** [comment]
I'll merge this if you explain what `certifi==2022.9.24
charset-normalizer==2.1.1
idna==3.4 urllib3==1.26.12 PyGObject-stubs==1.0.0` are for?

**Comment by ZappaBoy** [comment]
> I'll merge this if you explain what `certifi==2022.9.24 charset-normalizer==2.1.1 idna==3.4 urllib3==1.26.12 PyGObject-stubs==1.0.0` are for?

As I said in previous message I needed this packages to develop using IDE and the debugger. In particular I needed `PyGObject-stubs` to generate stubs in order to get suggestions from IDE. Feel free to remove them if you are not interested about that.

**Comment by deter0** [comment]
Okay thanks!

**Comment by dmolin** [comment]
I can confirm that this also fixes the problem I encountered with mpv 😄

**Comment by ZappaBoy** [comment]
> I can confirm that this also fixes the problem I encountered with mpv smile

Glad to hear it 👍🏼

**Comment by ZappaBoy** [comment]
Hi @deter0 , I noticed that this changes are reverted in 983c12371c94448f3d9a4687ee90779a6b156363 , what is the problem with this changes? It is possible to fix that?

**Comment by dmolin** [comment]
> Hi @deter0 , I noticed that this changes are reverted in [983c123](https://github.com/deter0/xborder/commit/983c12371c94448f3d9a4687ee90779a6b156363) , what is the problem with this changes? It is possible to fix that?

I confirm this is the case; the problem with the border is now present again...

**Pull request merged**